src/util/util.h

```diff
@@ -53,6 +53,10 @@
 #include "util/sss_format.h"
 #include "util/debug.h"
 
+#ifndef ALLPERMS
+#define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO)
+#endif
```

Comment:
On July 14, 2016 at 1:40:32 pm +02:00, lslebodn <<notifications@github.com>> wrote:

> In src/util/util.h <https://github.com/SSSD/sssd/pull/4#discussion_r70790794>:
> > @@ -53,6 +53,10 @@
> > #include "util/sss_format.h"
> > #include "util/debug.h"
> >
> > +#ifndef ALLPERMS
> > +#define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO)
> > +#endif
> 
> I can see ALLPERMS defined in sys/stat.h. (But I did't try with musl libc)
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/SSSD/sssd/pull/4/files/bd4c5186e85619c0d6fdc36690b311734e552fe4#r70790794>, or mute the thread <https://github.com/notifications/unsubscribe/AAipUsqhF0gIiR1CwCKj3_ZjLkE4v8DZks5qViAwgaJpZM4JMLmi>.
> 
> 
> 
> 
> Its not defined by the POSIX standard and as such, musl doesn't include it in sys/stat.h and therefor it has to be defined somewhere else, in this case I identified that the files that referenced the macro both included util.h and thus added it there.

Mvh
Daniel Kuehn, 073-181 577 2
Data@UrService
Hemsida: <http://www.dataaturservice.se>